projects/maps-ng/src/components/si-map/shared/themes/style.ts

```diff
@@ -17,6 +20,18 @@ const green300 = (opacity: number): string => `rgba(114, 230, 163, ${opacity})`;
 const blue700 = 'rgb(30, 82, 153)';
 const blue100 = 'rgb(210, 226, 247)';
 
+/** Indicate the current theme is dark otherwise light. */
+export const observeDarkTheme = (destroyRef?: DestroyRef): Signal<boolean> => {
+  destroyRef ??= inject(DestroyRef);
+  return toSignal(
+    fromEvent<CustomEvent>(window, 'theme-switch').pipe(
+      takeUntilDestroyed(destroyRef),
+      map(e => e.detail.dark)
+    ),
+    { initialValue: document.documentElement.classList.contains('app--dark') }
+  );
+};
+
```

Comment:
should move to separate .ts file since this isn't tied to our default style but a nice utility function